packages/labs/ssr-dom-shim/src/index.ts

```diff
@@ -54,11 +60,17 @@ const ElementShim = class Element {
   }
   attachShadow(init: ShadowRootInit): ShadowRoot {
     const shadowRoot = {host: this} as object as ShadowRoot;
+    this.__shadowRootMode = init.mode;
     if (init && init.mode === 'open') {
       this.__shadowRoot = shadowRoot;
     }
     return shadowRoot;
   }
+  attachInternals(): ElementInternals {
+    const internals = new InternalsShim(this as unknown as HTMLElement);
+    this.__internals = internals;
+    return internals as ElementInternals;
```

Comment:
Also getting TS errors on this assignment and casting on local IDE due to `ariaColIndexText, ariaRowIndexText`.

I wonder if the `InternalsShim` should get the type treatment like the other shims in this file:
```ts
const ElementShimWithRealType = ElementShim as object as typeof Element;
export {ElementShimWithRealType as Element};
```

Reply:
Yeah, they type `ARIAMixin` is apparently really fragile. I'll back out of that and use `string` instead. I think most of the issues you're seeing are related to that.

Reply:
`tsc` wasn't breaking so it turns out the issue was my VS Code was using TS v4.9.5 whereas the workspace's v4.7.4.

It seems `ariaColIndexText` and `ariaRowIndexText` were added to TS's `lib.dom.d.ts` in a later version as they're part of the spec, but in practice not implemented by browsers. MDN has also pulled them from their page for that reason https://github.com/mdn/browser-compat-data/pull/18546.

Since this builds, maybe okay to leave as is, but we could also `Omit<ARIAMixin, 'ariaColIndexText' | 'ariaRowIndexText'>` mentioning lack of support for future proofing.

Reply:
Actually.. `Omit<ARIAMixin, 'ariaColIndexText' | 'ariaRowIndexText'>` will help for `implements` but not for casting to built in `ElementInternals` type. So.. maybe okay for now, we can worry about it when we update TS.